FinSearchUnified/Resources/views/frontend/listing/filter/_includes/filter-color-selection.tpl

```diff
@@ -21,9 +21,13 @@
     <label class="filter-panel--label"
            for="__{$facet->getFieldName()|escape:'htmlall'}__{$option->getId()|escape:'htmlall'}">
 
-        {if $option->getColorcode()}
-            {$mediaFile = $option->getColorcode()}
-            <div class="color--filter-item" style="background: {$mediaFile};" title="{$option->getId()|escape:'htmlall'}"></div>
+        {$colorCode = $option->getColorcode()}
+        {$imageUrl = $option->getImageUrl()}
+
+        {if $colorCode && $imageUrl}
+            <div class="color--filter-item" style="background-image: url({$imageUrl}); background-color: {$colorCode};" title="{$option->getId()|escape:'htmlall'}"></div>
+        {elseif $colorCode}
+            <div class="color--filter-item" style="background: {$colorCode};" title="{$option->getId()|escape:'htmlall'}"></div>
         {else}
```

Comment:
Wouldn't this mean that if an image does not have a color code, but has a custom `$imageUrl`, that always the no-picture is shown?
In case it is not, those conditions are not really self-explanatory, can we make this a bit more clear?

Reply:
I added another if statement for `$imageUrl`. Should be more explanatory now.